src/diffusers/loaders.py

```diff
@@ -917,286 +984,152 @@ def load_lora_weights(self, pretrained_model_name_or_path_or_dict: Union[str, Di
+                            f"{name}.k_proj.lora_linear_layer.up.weight"
+                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_k_lora.up.weight")
+                        text_encoder_lora_state_dict[
+                            f"{name}.v_proj.lora_linear_layer.up.weight"
+                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_v_lora.up.weight")
+                        text_encoder_lora_state_dict[
+                            f"{name}.out_proj.lora_linear_layer.up.weight"
+                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_out_lora.up.weight")
+
+                        text_encoder_lora_state_dict[
+                            f"{name}.q_proj.lora_linear_layer.down.weight"
+                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_q_lora.down.weight")
+                        text_encoder_lora_state_dict[
+                            f"{name}.k_proj.lora_linear_layer.down.weight"
+                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_k_lora.down.weight")
+                        text_encoder_lora_state_dict[
+                            f"{name}.v_proj.lora_linear_layer.down.weight"
+                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_v_lora.down.weight")
+                        text_encoder_lora_state_dict[
+                            f"{name}.out_proj.lora_linear_layer.down.weight"
+                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_out_lora.down.weight")
+
+                rank = text_encoder_lora_state_dict[
+                    "text_model.encoder.layers.0.self_attn.out_proj.lora_linear_layer.up.weight"
+                ].shape[1]
+
+                cls._modify_text_encoder(text_encoder, lora_scale, network_alpha, rank=rank)
+
+                # set correct dtype & device
+                text_encoder_lora_state_dict = {
+                    k: v.to(device=text_encoder.device, dtype=text_encoder.dtype)
+                    for k, v in text_encoder_lora_state_dict.items()
+                }
+
+                load_state_dict_results = text_encoder.load_state_dict(text_encoder_lora_state_dict, strict=False)
+                if len(load_state_dict_results.unexpected_keys) != 0:
+                    raise ValueError(
+                        f"failed to load text encoder state dict, unexpected keys: {load_state_dict_results.unexpected_keys}"
+                    )
+
     @property
     def lora_scale(self) -> float:
         # property function that returns the lora scale which can be set at run time by the pipeline.
         # if _lora_scale has not been set, return 1
         return self._lora_scale if hasattr(self, "_lora_scale") else 1.0
 
-    @property
-    def text_encoder_lora_attn_procs(self):
-        if hasattr(self, "_text_encoder_lora_attn_procs"):
-            return self._text_encoder_lora_attn_procs
-        return
-
-    def _remove_text_encoder_monkey_patch(self):
```

Comment:
Actually this was so far the only way to remove the LoRA layer from the text encoder, so I assume lots of people are using it. 

So I think we need to gracefully deprecate this method instead of changing it to a class method (because the signature is changed).

Also could you maybe add a new cleaner "remove_lora()" the both removes all LoRA from the text encoder and also replaces the LoRA attn processor to its equivalent non LoRA attn processor?

Reply:
We introduced this only after https://github.com/huggingface/diffusers/pull/3437. So, I think it's okay to directly introduce this actually. The disruption will be contained given how new it is.

Reply:
If needed we can leave it as an instance method and just move the body to a class method. However, it's underscore prefixed so we should be able to assume it's private and we can change it if necessary?

Reply:
Hmm, it was in the last release though. Agree that it was underscored though. Actually @williamberman is it possible to do some kind of hack maybe to deprecate by doing:

```py
if type(cls) != type:
    # if cls is not a class but a `self` (instantiated class) we fallback to deprecated method
```

would something like this maybe work?

Reply:
Very janky hack, but I see your point. 

(The ^ statement is a funny banter and shouldn't be taken seriously)

Reply:
Can we make sure this is backwards comp?

Reply:
For a function marked as a classmethod, regardless of if the function is called on the class level or an instance, python will always pass the class to the first argument.

i.e.

```py
class Test:
    @classmethod
    def foo(cls):
        print(cls)

Test.foo()
Test().foo()
```

prints

```
<class '__main__.Test'>
<class '__main__.Test'>
```

The clearest thing to do here is leave the existing function as an instance method and have it call into a class method

i.e.

```py
    def _remove_text_encoder_monkey_patch(self):
        self._remove_text_encoder_monkey_patch_classmethod(self.text_encoder)

    @classmethod
    def _remove_text_encoder_monkey_patch_classmethod(cls, text_encoder):
        for _, attn_module in text_encoder_attn_modules(text_encoder):
            if isinstance(attn_module.q_proj, PatchedLoraProjection):
                attn_module.q_proj = attn_module.q_proj.regular_linear_layer
                attn_module.k_proj = attn_module.k_proj.regular_linear_layer
                attn_module.v_proj = attn_module.v_proj.regular_linear_layer
                attn_module.out_proj = attn_module.out_proj.regular_linear_layer
```

is that ok?

Reply:
This works for me. It's backward-compatible too. WDYT @patrickvonplaten?

Reply:
ok for me!

Reply:
added!